src/dmd/typesem.d

```diff
@@ -1494,7 +1494,10 @@ extern(C++) Type typeSemantic(Type type, const ref Loc loc, Scope* sc)
             // Now that we completed semantic for the argument types,
             // run semantic on their default values,
             // bearing in mind tuples have been expanded.
-            size_t tupleStartIdx = size_t.max;
+            // We need to keep a pair of [oidx, eidx] (original index,
+            // extended index), so we now to run semantic when the original
+            // index changes for the first time.
+            size_t[2] tupleStartIdx = [ size_t.max, size_t.max ];
```

Comment:
I don't see a purpose for making this an array, as it is never indexed. Might as well call it `tupleStartIdx1` and `tupleStartIdx2`. Or `tupleOrigIdx` and `tupleExtIdx`.